src/anyio/streams/memory.py

```diff
@@ -59,10 +62,17 @@ def statistics(self) -> MemoryObjectStreamStatistics:
 class MemoryObjectReceiveStream(Generic[T_co], ObjectReceiveStream[T_co]):
     _state: MemoryObjectStreamState[T_co]
     _closed: bool = field(init=False, default=False)
+    _source_traceback: traceback.StackSummary | None = field(init=False, default=None)
 
     def __post_init__(self) -> None:
         self._state.open_receive_channels += 1
 
+        if self.is_source_traceback_capturing_enabled():
+            self._source_traceback = traceback.extract_stack(sys._getframe(1))
+
+    def is_source_traceback_capturing_enabled(self) -> bool:
+        return sys.flags.dev_mode
```

Comment:
I'd rather not add a new public method for this.
```suggestion
        if sys.flags.dev_mode:
            self._source_traceback = traceback.extract_stack(sys._getframe(1))
```